File: backend/src/main/java/br/ufpr/sept/so2/modules/iam/api/dto/RecuperarSenhaRequest.java
package br.ufpr.sept.so2.modules.iam.api.dto;

import jakarta.validation.constraints.Email;
import jakarta.validation.constraints.NotBlank;

public record RecuperarSenhaRequest(
        @NotBlank(message = "Informe um e-mail")
        @Email(message = "Informe um e-mail válido")
        String email
) {
}

File: backend/src/main/kotlin/br/ufpr/sept/so2/modules/iam/api/dto/RecuperarSenhaRequest.kt
package br.ufpr.sept.so2.modules.iam.api.dto

import jakarta.validation.constraints.Email
import jakarta.validation.constraints.NotBlank

data class RecuperarSenhaRequest(
    @field:NotBlank(message = "Informe um e-mail")
    @field:Email(message = "Informe um e-mail válido")
    val email: String,
)
